Source/ScaleStructureController/ScaleStructureDialog.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+  ==============================================================================
+
+    ScaleDesignWindow.cpp
+    Created: 24 May 2020 7:09:24pm
+    Author:  Vincenzo
+
+  ==============================================================================
+*/
+
+#include "ScaleStructureDialog.h"
+
+ScaleDesignWindow::ScaleDesignWindow(ScaleStructure& scaleStructure, Array<Colour>& colourTable, Colour backgroundColour)
```

Comment:
Maybe entitle it “Scale/colour scheme designer”, since the colour scheme is important here.

Reply:
I don't mind "ScaleColourDesigner". The class name for the component is ScaleStructureComponent, but I'd like to use something a bit nicer for the instance.
Maybe the window class can be called ScaleColourDesignerWindow, and the component could be named scaleColourDesigner?

Reply:
Yes, maybe the component ScaleColourDesignerComponent. But actually the class names in the code are not even that important, we could also leave them. But the title of the window in the running program would be good to have the word “colour” in it.